alf/networks/encoding_networks.py

```diff
@@ -230,6 +230,149 @@ def __init__(self,
         super().__init__(nets, input_tensor_spec=input_tensor_spec, name=name)
 
 
+@alf.configurable
+class ImageDeconvNetwork(_Sequential):
+    """
+    A general template class for creating image deconv (transposed convolutional)
+        networks with auto-shape inference.
+    """
+
+    def __init__(self,
+                 input_size,
```

Comment:
I think now we start to use type hints gradually in the code?

Reply:
Added